spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/DenseTensor.scala

```diff
@@ -866,6 +866,38 @@ private[tensor] class DenseTensor[@specialized T: ClassTag](
     (values, indices)
   }
 
+  protected def sumNSquare(n: T): T = {
+    var res: T = ev.fromType(0)
+    val func = new TensorFunc2[T] {
+      override def apply(data1: Array[T], offset1: Int): Unit = {
+        res = ev.plus(res, ev.pow(ev.abs(data1(offset1)), n))
+      }
+    }
+    DenseTensorApply.apply1[T](this, func)
+    res
+  }
+
+//  override def sumSquare(): T = {
+//    sumNSquare(ev.fromType[Int](2))
+//  }
+
+  override def sumSquare(): T = {
+    this.dot(this)
+  }
+
+  override def clamp(min: Float, max: Float): Tensor[T] = {
+    val maxT = ev.fromType[Float](max)
+    val minT = ev.fromType[Float](min)
+    val func = new TensorFunc2[T] {
+      override def apply(data1: Array[T], offset1: Int): Unit = {
+        if (ev.isGreater(data1(offset1), maxT)) data1(offset1) = maxT
+        else if (ev.isGreater(minT, data1(offset1))) data1(offset1) = minT
+      }
```

Comment:
```scala
if (ev.isGreater(data1(offset1), maxT)) data1(offset1) = maxT
else if (ev.isGreater(minT, data1(offset1))) data1(offset1) = minT
```
could be 
```scala
ev.min(ev.max(data1(offset, minT)), maxT)
```

Reply:
The latter code is much cleaner than the previous one. But I think it may affect performance since it need compare twice every time